package/src/core/define-integration.ts

```diff
@@ -43,12 +43,15 @@ export const defineIntegration = <
 		name: string;
 		options: z.output<TOptionsSchema>;
 	}) => ExtendedHooks<TPlugins>;
-}): ((options?: z.input<TOptionsSchema>) => AstroIntegration) => {
-	return (_options: z.input<TOptionsSchema> = {}) => {
-		const parsedOptions = (optionsSchema || z.record(z.any())).safeParse(
-			_options,
-			{ errorMap },
-		);
+}): ((
+	...args: undefined extends z.input<TOptionsSchema>
+		? [options?: z.input<TOptionsSchema>]
+		: [options: z.input<TOptionsSchema>]
+) => AstroIntegration) => {
+	return (...args) => {
+		const parsedOptions = (optionsSchema ?? z.any()).safeParse(args[0], {
```

Comment:
@Fryuni do you think it would be better to fallback to z.never?

Reply:
Setting the option schema would be required to have any options. It's a matter of preference, I always prefer to have the schema so I think that would be great.